internal/controller/ols_app_redis_assets.go

```diff
@@ -56,10 +101,21 @@ func (r *OLSConfigReconciler) generateRedisDeployment(cr *olsv1alpha1.OLSConfig)
 									corev1.ResourceMemory: resource.MustParse("1Gi"),
 								},
 							},
+							Command: []string{"redis-server",
+								"--port", "0",
+								"--tls-port", "6379",
+								"--tls-cert-file", path.Join(OLSAppCertsMountRoot, "tls.crt"),
+								"--tls-key-file", path.Join(OLSAppCertsMountRoot, "tls.key"),
+								"--tls-ca-cert-file", path.Join(OLSAppCertsMountRoot, RedisCAVolume, "service-ca.crt"),
+								"--tls-auth-clients", "optional",
+								"--protected-mode", "no",
+								"--requirepass", redisPassword},
 						},
 					},
+					Volumes: volumes,
 				},
 			},
+			RevisionHistoryLimit: &revisionHistoryLimit,
```

Comment:
I don't know if we will use the rollback, but I see no reason to set this value to zero.

Reply:
If I do not set this to 0, whenever there is a pod restart their replica sets remain stale and unused as a history which was unclean as below.
```
NAME                                                   DESIRED   CURRENT   READY   AGE
replicaset.apps/lightspeed-app-server-66cdc9899f       1         1         1       34s
replicaset.apps/lightspeed-console-plugin-785b68cdcc   2         2         2       2m55s
replicaset.apps/lightspeed-redis-server-6895dfb689     1         1         1       34s
replicaset.apps/lightspeed-redis-server-6b5c6d8df      0         0         0       2m55s
replicaset.apps/lightspeed-redis-server-77d5455894     0         0         0       113s
```

Reply:
They're not exactly "stolen". 
The pod template configuration was updated so a new rs was created.
If the deployment fails during the update, OCP can roll back to the previous version.
I still think we should at least leave the value 1..3 (default values is 10)

Reply:
I am okay with that. But I will let @raptorsun and @xrajesh to chime in here to decide on an exact number.

Reply:
I think we should leave at least one redis ready when upgrading. Elsewise, the application server will panic due to disconnection from cache server.